DirectProgramming/C++SYCL_FPGA/Tutorials/Features/multi_aocx/dynamic_load/README.md

```diff
@@ -0,0 +1,275 @@
+| Software                          | Intel® oneAPI DPC++/C++ Compiler
+| What you will learn               | How to use the `dynamic_load` flow tests to split kernels into multiple different images to split up compile time.
+| Time to complete                  | 30 minutes
+
+> **Note**: Even though the Intel DPC++/C++ OneAPI compiler is enough to compile for emulation, generating reports and generating RTL, there are extra software requirements for the simulation flow and FPGA compiles.
+>
+> For using the simulator flow, one of the following simulators must be installed and accessible through your PATH:
+> - Questa*-Intel® FPGA Edition
+> - Questa*-Intel® FPGA Starter Edition
+> - ModelSim® SE
+>
+> When using the hardware compile flow, Intel® Quartus® Prime Pro Edition must be installed and accessible through your PATH.
+
+## Purpose
+
+The purpose of this tutorial is to describe the `dynamic_load` and to show how it can be used to compile kernels into shared libraries that can be loaded dynamically at runtime. This can be useful in generating small binaries, and only loading the relevant kernels into memory at runtime.
+
+### Using the dynamic_load flow
+
+`dynamic_load` flow allows you to create shared libraries for the kernels and loads only the desired libraries at runtime, therefore generates a smaller binary than `dynamic_link`.
+
+- It is similar to dynamic_link BUT Loads the shared lib with dlopen in host code to access.
+- Only loaded libraries are in memory therefore reducing memory usage.
+- However it requires cubersome extra steps to load libraries, functions and handle name mangling.
+- It is recommended for designs with many aocxs.
+
+`main.cpp`:
+```
+queue q;
+add_lib = dlopen("./add.so", RTLD_NOW);
+add = dlsym(add_lib, “add");
+mul_lib = dlopen("./mul.so", RTLD_NOW);
+mul = dlsym(mul_lib, “mul");
+add(q);
+mul(q);
+```
+
+`vector_add.cpp`:
+```
+extern “C” {
+  void add(queue q) {
+    q.submit();
+  }
+}
+```
+
+`vector_mul.cpp`:
+```
+extern “C” {
+  void mul(queue q) {
+    q.submit();
+  }
+}
+```
+
+```
+icpx ... -fPIC -c vector_mul.cpp -o mul.o
+icpx ... -fPIC -shared mul.o -o mul.so
+icpx ... -fPIC -c vector_add.cpp -o add.o
+icpx ... -fPIC -shared add.o -o add.so
+icpx ... main.cpp -o main.exe
+```
+Note: the `-fPIC` flag is used to generate Position Independent Code, which is a binary code for both static library and share library. It allows a shared library to statically link to zlib.
+
+## Key Concepts
+The `dynamic load` flow method requires the user to separate the host and device code into separate files. You can choose dynamically at runtime which libraries to load and only the loaded libraries are in memory. However this requires extra steps to load, therefore only recommended for design with many aocxs.
```

Comment:
Looks good - one last comment. Are oneAPI users exposed to aocx images or should we use a different terminology here?